src/utils/__tests__/quotesService.test.js

```diff
@@ -3,75 +3,50 @@ import { BRAG, CONFESS } from '../../constants/roleActions';
-  DEVOPS_BRAG: {
-    '1': 'DevOps-Brag test quote no. 1',
-    '2': 'DevOps-Brag test quote no. 2'
-  },
-  DEVOPS_CONFESS: {
-    '1': 'DevOps-Confess test quote no. 1',
-    '2': 'DevOps-Confess test quote no. 2'
-  }
-}));
-
-describe('quotesService - getQuote', () => {
-  it('should return the quote for BackEnd Brag with quoteID = 1', () => {
-    const quote = getQuote(BACK_END_ROLE, BRAG, '1');
-
-    expect(quote).toEqual('Backend-Brag test quote no. 1');
-  });
-
-  it('should return the quote for BackEnd Brag with quoteID = 2', () => {
-    const quote = getQuote(BACK_END_ROLE, BRAG, '2');
-
-    expect(quote).toEqual('Backend-Brag test quote no. 2');
-  });
-
-  it('should return the quote for FrontEnd Confess with quoteID = 1', () => {
-    const quote = getQuote(FRONT_END_ROLE, CONFESS, '1');
-
-    expect(quote).toEqual('FrontEnd-Confess test quote no. 1');
-  });
-
-  it('should return the quote for DevOps Confess with quoteID = 1', () => {
-    const quote = getQuote(DEV_OPS_ROLE, CONFESS, '1');
-
-    expect(quote).toEqual('DevOps-Confess test quote no. 1');
-  });
-
-  it('should return the default quote for not defined role, action or quoteID', () => {
-    const quote = getQuote(null, null, null);
-
-    expect(quote).toEqual("Daily in 5 minutes and I'm still not sure what to say...");
+let count_chars = (str, char) => str.split('').reduce((acc, ch) => (ch === char ? acc + 1 : acc), 0);
+
+describe('quotesService - getRandomQuoteID should return something long enough with spacers', () => {
+  [BACK_END_ROLE, FRONT_END_ROLE, DEV_OPS_ROLE].map(role => {
+    it('should return random possible quoteID for backend role and BRAG', () => {
+      const randomQuoteID = getRandomQuoteID(role, BRAG);
+      expect(count_chars(randomQuoteID, '-')).toEqual(3);
+      expect(randomQuoteID.length).toBeGreaterThan(3 * 4);
+    });
+    it('should return random possible quoteID for backend role and BRAG', () => {
+      const randomQuoteID = getRandomQuoteID(role, CONFESS);
+      expect(count_chars(randomQuoteID, '-')).toEqual(3);
+      expect(randomQuoteID.length).toBeGreaterThan(3 * 4);
+    });
   });
 });
 
-describe('quotesService - getRandomQuoteID', () => {
-  it('should return random possible quoteID for given action and role', () => {
-    const randomQuoteID = getRandomQuoteID(BACK_END_ROLE, BRAG);
-
-    expect(['1', '2']).toContain(randomQuoteID);
+describe('quotesService - getRandomQuoteID and then verify if text is the same', () => {
+  [BACK_END_ROLE, FRONT_END_ROLE, DEV_OPS_ROLE].map(role => {
+    it('given 4 trials it should have chances of 576 ** 4 to be all the same', () => {
+      // this is probability but given several iterations the list should be different
+      let x = new Set();
+      x.add(getRandomQuoteID(role, BRAG));
+      x.add(getRandomQuoteID(role, BRAG));
+      x.add(getRandomQuoteID(role, BRAG));
+      x.add(getRandomQuoteID(role, BRAG));
+      expect(x.size).toBeGreaterThan(1);
```

Comment:
I don't get this test. Can you elaborate what it does?

Reply:
Yes, It generates sufficiently large sample of random quotes and ensures they are not the same.
If bug occurs it would very likely generate the same result, so we expect that the outcome is various.